### PR TITLE
Added ability to set umask for file_system and php_file driver

### PR DESCRIPTION
**Pull request by yethee**
These changes add support doctrine/cache of version 1.4.2.

### CODE REVIEW

**Comment by xabbuh** [comment]
What about bumping the required `doctrine/cache` version in `composer.json` too?

**Comment by yethee** [comment]
Done.

File: composer.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
         "symfony/security":        "~2.2",
         "symfony/doctrine-bridge": "~2.2",
         "doctrine/inflector":      "~1.0",
-        "doctrine/cache":          "~1.3"
+        "doctrine/cache":          "~1.4.2"
```
Comment (xabbuh) [inline composer.json]: This must be `^1.4.2`. Otherwise you would not allow anything starting with `1.5` anymore.

Comment (yethee) [inline composer.json]: You are right. Fixed.

**Pull request merged**